### PR TITLE
Add parted module

### PR DESCRIPTION
**Pull request by kozhukalov**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

parted
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel 5fdcc2f2b0) last updated 2016/06/24 16:54:40 (GMT +300)
  lib/ansible/modules/core: (detached HEAD 4fe583e29b) last updated 2016/06/24 16:55:07 (GMT +300)
  lib/ansible/modules/extras: (devel a3c4daa63c) last updated 2016/07/04 16:03:40 (GMT +300)
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides

```
##### SUMMARY

This new module parted allows to run parted scripts as well as gather partition information

This module allows to run parted scripts.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @kozhukalov for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by kozhukalov** [comment]
@gregdek Could you please help me understand what 'revisions' is and where I can get it?

**Comment by kozhukalov** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @kozhukalov for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: system/parted.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+options:
+  device:
+    description:
+    - The device file. E.g. /dev/sda
+    required: true
+  align:
+    description:
+    - Alignment type for newly created partitions
+    required: false
+    choices: ["none", "cylinder", "minimal", "optimal"]
```
Comment (resmo) [inline system/parted.py:42]: the default doc is missing according your arg_spec:

```
default: optimal
```

File: system/parted.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+            'fstype': part_params[4] or None
+        })
+    return {'generic': generic, 'partitions': parts}
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            device=dict(required=True),
+            align=dict(default='optimal'),
```
Comment (resmo) [inline system/parted.py]: add the choices list like you did in the arg docs:

``` python
align=dict(default='optimal', choices=['none', 'cylinder', 'minimal', 'optimal']),
```

File: system/parted.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    partitions:
+      description: List of device partitions
+      type: list
+'''
+
+
+EXAMPLES = '''
+# Do nothing but get partition info.
+- parted:
+  device: /dev/sda
```
Comment (resmo) [inline system/parted.py]: missing indentation in all of your examples

**Comment by resmo** [comment]
hi @kozhukalov thanks for this module, a few minor tweeks and we are good to go.

**Comment by mattclay** [comment]
Tests on shippable failed because this PR branch was created before shippable was enabled. If you rebase the branch the tests will pass, although in this case the tests will just no-op since there are no tests for this module.

**Comment by mattclay** [comment]
Ignore my previous comment, it looks like the branch has already been rebased.

**Comment by kozhukalov** [comment]
ready_for_review

**Comment by miniBill** [comment]
What's the status of this PR?

**Comment by resmo** [comment]
The module is not idempotent, it seem just like to be a wrapper around parted.

**Comment by kozhukalov** [comment]
Yes, it is not idempotent. Will implement idempotency.

**Comment by resmo** [comment]
needs_revision

**Comment by gregdek** [comment]
Thanks @kozhukalov for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@kozhukalov A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by waddles** [comment]
I'm keen to see idempotency implemented.

**Comment by gregdek** [comment]
@kozhukalov Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by lhoss** [comment]
we are very interested in this module as well 👍  (to replace our `parted` shell tasks)
let's hope @kozhukalov  finds some time for the final review fixes .. else others (like us) could take over !

**Comment by gregdek** [comment]
@kozhukalov A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@kozhukalov Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@kozhukalov A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Closing due to contributor inactivity -- @lhoss maybe you could fork and continue the work here?

**Comment by grugnog** [comment]
If anyone is considering working on this, I would suggest considering sgdisk in place of parted. I just had to handle some partition tasks with Ansible and found it much easier to both script and parse than parted.

**Pull request closed without merging**